Sources/PackageCollections/PackageCollections.swift

```diff
@@ -14,6 +14,13 @@ import TSCBasic
 
 // TODO: is there a better name? this conflicts with the module name which is okay in this case but not ideal in Swift
 public struct PackageCollections: PackageCollectionsProtocol {
+    // Check JSONPackageCollectionProvider.isSignatureCheckSupported before updating or removing this
```

Comment:
👍

Reply:
I wonder if there is a way to exclude the entire PackageCollections module in unsupported platforms. not sure if easier / harder but if easy may hep to keep the code simpler and move the complexity there

Reply:
cc @neonichu may know of a way

Reply:
In principle, we could use [conditional dependencies](https://github.com/apple/swift-evolution/blob/main/proposals/0273-swiftpm-conditional-target-dependencies.md#proposed-solution), might need some reorganization because I think right now some of the collections targets are also directly included in products.